src/scheduler/job_info.c

```diff
@@ -4248,7 +4248,7 @@ queue_subjob(resource_resv *array, server_info *sinfo,
 	if (subjob_index >= 0) {
 		subjob_name = create_subjob_name(array->name, subjob_index);
 		if (subjob_name != NULL) {
-			if ((rresv = find_resource_resv(sinfo->jobs, subjob_name)) != NULL) {
+			if ((rresv = find_resource_resv(qinfo->jobs, subjob_name)) != NULL) {
```

Comment:
Just a quick comment: While this might look right, it is not.  The point of this check is to see if the job is in the system.  You want to search the entire system, not just a subset.

Reply:
Was just testing the water. I thought of speeding this up since array jobs do not span across queues from server's point of view. But now after looking at movejob server code I realize there are possibilities of subjob instances internally spanning across queues (moving partially completed array job to a different queue)

However I feel that there is an opportunity to speed things up here, as we are scanning the whole system job pool (O(n)) before creation of each new subjob.
Anyways since this is not needed for the problem at hand, I will revert this change and leave it to the sched owner's discretion on the speed up opportunity.

Reply:
Thanks for thinking of scheduler speed.  This is something that is always in the back of our minds.

Reply:
This change now stands reverted.